src/cmd/cli/command/stack.go

```diff
@@ -0,0 +1,174 @@
+					Options: []string{"AWS", "GCP", "Azure"},
+					Default: "AWS",
+				}, &provider, survey.WithStdio(term.DefaultTerm.Stdio()))
+				if err != nil {
+					return err
+				}
+
+				err = providerID.Set(provider)
+				if err != nil {
+					return err
+				}
+				params.Provider = providerID
+			}
+
+			if params.Region == "" {
+				defaultRegion := ""
+				switch providerID {
+				case cliClient.ProviderAWS:
+					defaultRegion = "us-west-2"
+				case cliClient.ProviderGCP:
+					defaultRegion = "us-central1"
+				}
+
+				region := ""
+
+				err := survey.AskOne(&survey.Input{
+					Message: "Enter cloud region for the stack deployment:",
+					Default: defaultRegion,
+				}, &region, survey.WithStdio(term.DefaultTerm.Stdio()))
+				if err != nil {
+					return err
+				}
+
+				params.Region = region
+			}
+
+			if params.Mode == modes.ModeUnspecified {
+				selectedMode := ""
+				err := survey.AskOne(&survey.Select{
+					Message: "Select deployment mode:",
+					Options: modes.AllDeploymentModes(),
+					Default: modes.ModeAffordable.String(),
+				},
+					&selectedMode, survey.WithStdio(term.DefaultTerm.Stdio()))
+				if err != nil {
+					return err
+				}
+
+				modeParsed, err := modes.Parse(selectedMode)
+				if err != nil {
+					return err
+				}
+				params.Mode = modeParsed
+			}
+
+			term.Debugf("Creating stack with parameters: %+v\n", params)
+
+			return stacks.Create(params)
+		},
+	}
+	stackNewCmd.Flags().VarP(&mode, "mode", "m", fmt.Sprintf("deployment mode; one of %v", modes.AllDeploymentModes()))
+	stackNewCmd.Flags().StringP("region", "r", "", "Cloud region for the stack deployment")
+
+	return stackNewCmd
+}
+
+func makeStackListCmd() *cobra.Command {
+	stackListCmd := &cobra.Command{
+		Use:     "list",
+		Aliases: []string{"ls"},
+		Args:    cobra.NoArgs,
+		Short:   "List existing Defang deployment stacks",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			jsonMode, _ := cmd.Flags().GetBool("json")
+			stacks, err := stacks.List()
+			if err != nil {
+				return err
+			}
+
+			if jsonMode {
+				jsonData := []byte("[]")
+				if len(stacks) > 0 {
+					jsonData, err = json.MarshalIndent(stacks, "", "  ")
+					if err != nil {
+						return err
+					}
+				}
+
+				_, err = term.Print(string(jsonData) + "\n")
+				return err
+			}
+
+			if len(stacks) == 0 {
+				_, err = term.Infof("No Defang stacks found in the current directory.\n")
+				return err
+			}
+
+			return term.Table(stacks, []string{"Name", "Provider", "Region", "Mode"})
+		},
+	}
+	stackListCmd.Flags().Bool("json", false, "Output in JSON format")
+	return stackListCmd
+}
+
+func makeStackRemoveCmd() *cobra.Command {
+	var stackRemoveCmd = &cobra.Command{
+		Use:     "remove STACK_NAME",
+		Aliases: []string{"rm"},
+		Args:    cobra.ExactArgs(1),
+		Short:   "Remove an existing Defang deployment stack",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			name := args[0]
+			return stacks.Remove(name)
```

Comment:
I'm not sold on the usefulness of this command, other than symmetry. Should we add an error or warning if there's stuff deployed with this stack?